src/__tests__/components/NavBar/ShoppingCart.test.tsx

```diff
@@ -22,7 +23,9 @@ describe("ShoppingCart", () => {
   };
 
   it("the circle element should not be rendered", () => {
-    render(<ShoppingCart quantity={props2.quantity} loginState={props2.loginState} />);
+    renderWithMockedProvider(
```

Comment:
Could you please explain the reason for using mock data? Thank you!

Reply:
We used the renderWithMockedProvider since we initially tried to test the ShoppingCart icon on navbar interaction. The cart icon click event needs to involve the redux store, thus, we used the mock provider.  After the discussion with Tato and Blue, the test I mentioned should not be involved in this stage. Using the simple "render" is enough in those tests.